### ISSUE TITLE
java-exec Geneator ignores additional files during image building.

### ISSUE THREAD

**Issue by rafaeltuelho** [fabric8io/fabric8-maven-plugin#662]
### Description
According to f-m-p generators docs [1], you can include additional files inside the final image build by the f-m-p. The docs states:

 > "You can add additional files to the target image within baseDir by putting files into src/main/fabric8-include. These will be added with mode 0644, everything in src/main/fabric8-include/bin will be added with 0755."

But, testing this functionality I noticed that it only works when your app is considered fatJar. In my case I specified a mainClass. Following the generator's logic (https://github.com/fabric8io/fabric8-maven-plugin/blob/master/generator/java-exec/src/main/java/io/fabric8/maven/generator/javaexec/JavaExecGenerator.java#L161-L182)

... my case is not a fatJar if I specify a mainClass.

I believe the possibility of include additional files should be very useful for any kind of image builder/generator. Some times you need/want to include some additional files and this would be a very useful hack avoiding to create a whole custom assembly definition for this.

### Info

* f-m-p version : 3.2.1
* Maven version (`mvn -v`) : 3.3.9
```

```
* Kubernetes / OpenShift setup and version : minikube version: v0.12.2
* If it's a bug, how to reproduce :
* If it's a feature request, what is your use case : be able to include additional Files inside the image being generated by the f-m-p generators
* Sample project : *[GitHub Clone URL]*

**Comment by davsclaus** [comment]
Yeah that sounds a good idea to do that for plain main classes too. 

Anyone is much welcome to work on a PR

**Comment by davsclaus** [comment]
Yeah this is still outstanding

This code is only for fat jar
https://github.com/fabric8io/fabric8-maven-plugin/blob/master/generator/java-exec/src/main/java/io/fabric8/maven/generator/javaexec/JavaExecGenerator.java#L194

**Comment by rhuss** [comment]
We should check whether this is still valid and fix it in case.
